packages-internal/test-utils/src/createRenderer.tsx

```diff
@@ -740,5 +740,5 @@ function act<T>(callback: () => void | T | Promise<T>) {
 const bodyBoundQueries = within(document.body, { ...queries, ...customQueries });
 
 export * from '@testing-library/react/pure';
-export { act, cleanup, fireEvent };
+export { act, cleanup, fireEvent, rtlUserEvent };
```

Comment:
Or do you think that calling the setup sooner is a cleaner solution?
```diff
diff --git a/packages-internal/test-utils/src/createRenderer.tsx b/packages-internal/test-utils/src/createRenderer.tsx
index f91a019495..28c749ed16 100644
--- a/packages-internal/test-utils/src/createRenderer.tsx
+++ b/packages-internal/test-utils/src/createRenderer.tsx
@@ -17,9 +17,11 @@ import {
   screen as rtlScreen,
   Screen,
 } from '@testing-library/react/pure';
-import { userEvent as rtlUserEvent } from '@testing-library/user-event';
+import { userEvent, UserEvent } from '@testing-library/user-event';
 import { useFakeTimers } from 'sinon';
 
+const userEventSetup = userEvent.setup();
+
 interface Interaction {
   id: number;
   name: string;
@@ -271,7 +273,7 @@ interface ServerRenderConfiguration extends RenderConfiguration {
 export type RenderOptions = Partial<RenderConfiguration>;
 
 export interface MuiRenderResult extends RenderResult<typeof queries & typeof customQueries> {
-  user: ReturnType<typeof rtlUserEvent.setup>;
+  user: UserEvent;
   forceUpdate(): void;
   /**
    * convenience helper. Better than repeating all props.
@@ -300,7 +302,7 @@ function render(
   );
   const result: MuiRenderResult = {
     ...testingLibraryRenderResult,
-    user: rtlUserEvent.setup(),
+    user: userEventSetup,
     forceUpdate() {
       traceSync('forceUpdate', () =>
         testingLibraryRenderResult.rerender(
@@ -740,5 +742,5 @@ function act<T>(callback: () => void | T | Promise<T>) {
 const bodyBoundQueries = within(document.body, { ...queries, ...customQueries });
 
 export * from '@testing-library/react/pure';
-export { act, cleanup, fireEvent, rtlUserEvent };
+export { act, cleanup, fireEvent };
 export const screen: Screen = { ...rtlScreen, ...bodyBoundQueries };
```

Reply:
As you have added this in, WDYT @romgrk? 🤔

Reply:
Why does it need to be called sooner? Is `setup` broken?

Reply:
To be able to do this in the `beforeEach`: https://github.com/mui/mui-x/pull/13860/files#diff-8aa6298bf92e4f4db9af052e311cc939fc5ff10e9387b1a51479c71a92230661R67

Reply:
How do you feel about removing `beforeEach` and instead calling

```tsx
writeText = spy(navigator.clipboard, 'writeText');
```

in the test body, keeping the `afterEach` but with a `writeText?.restore();`?

That way the mocks are only set up for those tests that need it, and the user event setup can stay where it is.

Reply:
Good point. I'll look into that approach and see if it's not too tedious. 👌 
Didn't see the suggested usage included the recommendation for localized usage. 🙈

Reply:
Nothing to add here, I set it up that way because of the RTL docs linked above by Jan.

Reply:
Thank you for your input.
The suggested approach solves the problem without too much effort: https://github.com/mui/mui-x/pull/13860/commits/6f0dd5ef80814b921975e0f82fa18d62b21b3b9f
The browser tests still need attention, but the main ones that were affected by this problem are green. ✅ 👌